rewrite-gradle/src/main/java/org/openrewrite/gradle/UpdateGradleWrapperFiles.java

```diff
@@ -0,0 +1,498 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.gradle;
+
+import lombok.*;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.*;
+import org.openrewrite.gradle.search.FindGradleProject;
+import org.openrewrite.gradle.util.DistributionInfos;
+import org.openrewrite.gradle.util.GradleWrapper;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.marker.BuildTool;
+import org.openrewrite.marker.Markers;
+import org.openrewrite.properties.PropertiesParser;
+import org.openrewrite.properties.PropertiesVisitor;
+import org.openrewrite.properties.search.FindProperties;
+import org.openrewrite.properties.tree.Properties;
+import org.openrewrite.quark.Quark;
+import org.openrewrite.remote.Remote;
+import org.openrewrite.semver.ExactVersion;
+import org.openrewrite.semver.Semver;
+import org.openrewrite.semver.VersionComparator;
+import org.openrewrite.text.PlainText;
+
+import java.time.ZonedDateTime;
+import java.util.*;
+
+import static java.util.Objects.requireNonNull;
+import static org.openrewrite.PathUtils.equalIgnoringSeparators;
+import static org.openrewrite.gradle.util.GradleWrapper.*;
+import static org.openrewrite.internal.StringUtils.isBlank;
+
+@RequiredArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@EqualsAndHashCode(callSuper = false)
+public class UpdateGradleWrapperFiles extends ScanningRecipe<UpdateGradleWrapperFiles.GradleWrapperState> {
+
+    @Override
+    public String getDisplayName() {
+        return "Update Gradle wrapper";
+    }
+
+    @Override
+    public String getDescription() {
```

Comment:
For the name and description of this class, I'm not certain how to different it in a meaningful way to users you aren't familiar with the implementation. Is there a way we can add the recipes to the jar but not have the available to be executed directly?